### ISSUE TITLE
Type templates cannot implement interfaces

### ISSUE THREAD

**Issue by scolsen** [carp-lang/Carp#860]
Templates generated by struct types currently cannot implement interfaces and calling implements on them silently fails to register the template function with the interface--enhancing the compiler to allow this would make it easier to define generic functions over types.

```
鲤 :i qux
qux : (Fn [(f a)] a) = {
    
}
鲤 (deftype (Q a b) [this a that b])
鲤 (implements qux Q.this)
鲤 :i qux
qux : (Fn [(f a)] a) = {
    
}
Defined at line 1, column 15 in 'REPL'
鲤 :i Q.this
Q.this : (Fn [(Ref (Q a b) q)] (Ref a q))
Defined at unknown line, unknown column in 'Q.this template'
Documentation: gets the `this` property of a `Q`.
鲤 (meta Q.this "implements")
=> (qux)
鲤 
鲤 (qux (Q.init 1 2))
I can’t find any implementation for the interface `qux` of type (Fn [(r8 Int)] Int) at no info.

None of the possibilities have the correct signature:
     at dummy-file:0:0.
鲤 
```

**Comment by scolsen** [comment]
Current workaround-- you need to define a `defn` alias for the getter:

```

鲤 (implements qux Q-alias)
[INTERFACE ERROR] Q-alias : (Fn [(Ref (Q a b) c)] (Ref a c)) doesn't match the interface signature (Fn [(f a)] a)
[INTERFACE ERROR] Q-alias : (Fn [(Ref (Q a b) c)] (Ref a c)) doesn't match the interface signature (Fn [(f a)] a) at REPL:1:13.

Traceback:
(implements qux Q-alias) at REPL:1:1.

鲤 

```

**Comment by scolsen** [comment]
Oddly enough, we already have code in place that should handle the template cases and register them correctly--so something else is going on.

**Comment by scolsen** [comment]
Possibly the accessor functions I'm trying to register resolve to `Instantiate` objs, which we presently don't handle

**Comment by scolsen** [comment]
Fixed by #863
